nautobot/virtualization/forms.py

```diff
@@ -449,6 +449,43 @@ def __init__(self, *args, **kwargs):
             self.fields["untagged_vlan"].widget.add_query_param("location_id", location.pk)
             self.fields["tagged_vlans"].widget.add_query_param("location_id", location.pk)
 
+    def clean(self):
+        super().clean()
+        ip_addresses = self.cleaned_data.get("ip_addresses")
+        # If there is no ip_addresses field here
+        # we know it is a CreateForm
+        if ip_addresses is None:
+            pass
+        # EditForm Primary IP validation logic.
+        else:
+            vm = self.cleaned_data.get("virtual machine")
+            name = self.cleaned_data.get("name")
+            # Virtual Machine and name are guaranteed to have an unique VMInterface
+            vm_interface = VMInterface.objects.get(virtual_machine=vm, name=name)
+            # Check if the ip address exist on other VMInterfaces that are assigned to the virtual machine.
+            other_assignments_ip4_exist = IPAddressToInterface.objects.filter(
+                vm_interface__virtual_machine=vm, ip_address=vm.primary_ip4
+            ).exclude(vm_interface=vm_interface)
+            other_assignments_ip6_exist = IPAddressToInterface.objects.filter(
+                vm_interface__virtual_machine=vm, ip_address=vm.primary_ip6
+            ).exclude(vm_interface=vm_interface)
+            # IP address validation
+            # We do have the pre_delete signal ip_address_to_interface_pre_delete_validation
+            # to handle this scenario, but we want the ValidationError to bubble up on the form.
+            if vm:
+                if vm.primary_ip4 and vm.primary_ip4 not in ip_addresses and not other_assignments_ip4_exist:
+                    raise ValidationError(
+                        {
+                            "ip_addresses": f"Cannot remove IP address {vm.primary_ip4} from interface {vm_interface} on Virtual Machine {vm.name} because it is marked as its primary IPv{vm.primary_ip4.family} address"
+                        }
+                    )
+                if vm.primary_ip6 and vm.primary_ip6 not in ip_addresses and not other_assignments_ip6_exist:
+                    raise ValidationError(
+                        {
+                            "ip_addresses": f"Cannot remove IP address {vm.primary_ip4} from interface {vm_interface} on Virtual Machine {vm.name} because it is marked as its primary IPv{vm.primary_ip6.family} address"
```

Comment:
```suggestion
                            "ip_addresses": f"Cannot remove IP address {vm.primary_ip6} from interface {vm_interface} on Virtual Machine {vm.name} because it is marked as its primary IPv6 address"
```